qutebrowser/utils/log.py

```diff
@@ -448,13 +448,15 @@ def emit(self, record):
             # We don't log VDEBUG to RAM.
             self._data.append(record)
 
-    def dump_log(self, html=False):
+    def dump_log(self, html=False, level='vdebug'):
         """Dump the complete formatted log data as as string.
 
         FIXME: We should do all the HTML formatter via jinja2.
         (probably obsolete when moving to a widget for logging,
         https://github.com/The-Compiler/qutebrowser/issues/34
         """
+        # pylint: disable=protected-access
+        minlevel = logging._nameToLevel.get(level.upper(), VDEBUG_LEVEL)
```

Comment:
This is somewhat problematic - if I recall correctly, `logging._nameToLevel` *is* something which got changed in a recent python version, and it can break at any time as it's "private".

I can see two possibilities here:

- Use `getattr(logging, level.upper(), VDEBUG_LEVEL)` - however you'd need to validate the level yourself (otherwise someone could crash qutebrowser by using e.g. `basic_format` as level)
- Add our own dict from logging names to levels to `qutebrowser.utils.log`. Since there's only a handful of levels and it's unlikely that'll ever change, I think that'd be the nicer option.